setup.php

```diff
@@ -75,39 +43,81 @@
         $configured = false;
     }
 
-	$file_written = false;
+?>
 
-    if(is_writeable($configfile) && !$configured){
+<?php if($configured): //TODO, should this reveal exactly how to reconfigure? I don't this it would be a security issue ?>
+    <h2>System already configured</h2>
+<?php else: ?>
 
-		$handle = fopen($configfile, 'w');
+    <?php if(!function_exists('random_bytes') && !function_exists('openssl_random_pseudo_bytes')): ?>
```

Comment:
Should we test for `hash_equals` too?

That is what protects us from timing attacks so while it has no effect on the key generation it is definitely important for how secure running Selfauth down the line is.